reporter/src/web-app/js/src/components/TreeView.js

```diff
@@ -32,126 +36,229 @@ class TreeView extends React.Component {
-                autoExpandParent: true,
-                search: searchVal,
-            });
-        }
-    }
+  onExpand = (expandedKeys) => {
+      this.setState({
+          expandedKeys,
+          autoExpandParent: false,
+      });
+  }
 
-    renderTreeNode = (treeLeaf) => {
-        const { name } = treeLeaf;
-        const { search } = this.state;
-
-        let nameMapped = (
-            <span>
-                {name}
-            </span>
-        );
-
-        if (search && name.indexOf(search) >= 0) {
-            const index = name.indexOf(search);
-            const pre = name.substring(0, index);
-            const post = name.substring(index + search.length);
-            nameMapped = (
-                <span>
-                    {pre}
-                    <span
-                        style={{
-                            color: 'red',
-                        }}
-                    >
-                        {search}
-                    </span>
-                    {post}
-                </span>
-            );
-        }
-
-        const tooltip = (
-            treeLeaf.path.length >= 1 && (
-                <div className="tooltip-list">
-                    <div class-name="tooltip-item">
-                        {treeLeaf.path.slice(0).join(' / ')}
-                    </div>
-                </div>
-            )
-        );
-
-        const nameWithTooltip = (
-            <Tooltip overlayClassName="oss-tree-tooltip" placement="right" title={tooltip}>
-                {nameMapped}
-            </Tooltip>
-        );
-
-        return (
-            <TreeNode
-                key={treeLeaf.id || treeLeaf.name}
-                title={nameWithTooltip}
-            >
-                {treeLeaf.children.map(childLeaf => this.renderTreeNode(childLeaf))}
-            </TreeNode>
-        );
-    }
+  componentDidUpdate(_, prevState) {
+      const { search } = this.state;
 
-    render() {
-        const { tree, expandedKeys, autoExpandParent } = this.state;
-        return tree && (
-            <div>
-                <Search style={{ marginBottom: 8 }} placeholder="Search" onChange={this.onSearchChange} />
-                <Tree
-                    onExpand={this.onExpand}
-                    showLine
-                    autoExpandParent={autoExpandParent}
-                    expandedKeys={expandedKeys}
-                >
-                    {tree.map(rootLeaf => this.renderTreeNode(rootLeaf))}
-                </Tree>
-            </div>
-        );
-    }
+      if (prevState.search !== search) {
+          this.calcSorted();
+      }
+  }
+
+  calcSorted = () => {
+      const { search } = this.state;
+      const selectedNodes = search ? Object.keys(this.treeNodes).filter(node => this.treeNodes[node].name.indexOf(search) >= 0) : [];
+      const sorted = selectedNodes.sort((a, b) => {
+          const aIndexes = a.split('-');
+          const bIndexes = b.split('-');
+          let diff = 0;
```

Comment:
Propose to move diff=0 into for-loop on line 84

Reply:
moved